guide/src/ecosystem/logging.md

```diff
@@ -57,11 +57,39 @@ possible to satisfy, read the documentation about [caching].
 
 ## The Python to Rust direction
 
-To best of our knowledge nobody implemented the reverse direction yet, though it
-should be possible. If interested, the `pyo3` community would be happy to
-provide guidance.
+To have python logs be handled by Rust, one need only register a rust function to handle logs emitted from the core python logging module. 
+
+This has been implemented within the [pyo3-pylogger] crate. 
+```
+use log::{info, warn};
+use pyo3::prelude::*;
+fn main() {
+    // register the host handler with python logger, providing a logger target
+    // this does have to be set, but doesn't have to be different from your application name.
+    pyo3_pylogger::register("example_application_py_logger");
+
+    // initialize up a logger
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("trace")).init();
+
+    //just show the logger working from Rust.
+    info!("Just some normal information!");
+    warn!("Something spooky happened!");
+
+    // Ask pyo3 to set up embedded Python interpreter
+    pyo3::prepare_freethreaded_python();
+    Python::with_gil(|py| {
+        // Python code can now `import logging` as usual
+        py.run("import logging", None, None).unwrap();
+        // Log messages are forwarded to the rust handler
+        py.run("logging.error('Something bad happened')", None, None)
+            .unwrap();
+    });
+}
+```
```

Comment:
Some suggestions to make the example more concise:

```suggestion
```rust
use log::{info, warn};
use pyo3::prelude::*;

fn main() -> PyResult<()> {
    // register the host handler with python logger, providing a logger target
    // set the name here to something appropriate for your application
    pyo3_pylogger::register("example_application_py_logger");

    // initialize up a logger
    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("trace")).init();

    // Log some messages from Rust.
    info!("Just some normal information!");
    warn!("Something spooky happened!");

    // Log some messages from Python
    Python::with_gil(|py| {
        py.run(
            "
import logging
logging.error('Something bad happened')
",
            None,
            None,
        )
    })
}
```
```